testsuite/features/step_definitions/navigation_steps.rb

```diff
@@ -1012,3 +1016,67 @@
     # ignored
   end
 end
+
+# Navigate to first system listed
+When(/^I select the first system in the list$/) do
+  step %(I follow the left menu "Systems > Overview")
+  within(:xpath, '//section') do
+    row = find(:xpath, "//div[@class='table-responsive']/table/tbody/tr[.//td]", match: :first)
+    # click first link in first row
+    within(row) do
+      first('a').click
+    end
+  end
+end
+
+# Change first system's property
+When(/^I have a system with "([^"]*)" as "([^"]*)" property$/) do |property_value, property_name|
+  # change system's property and clean the search index to make sure everything's correctly updated
+  steps %(
+    When I select the first system in the list
+    And I follow "Properties" in the content area
+    And I enter "#{property_value}" as "#{property_name}"
+    And I click on "Update Properties"
+    Then I should see a "System properties changed" text
+    And I clean the search index on the server
+    And I wait for "5" seconds
+        )
+end
+
+# Change first system's property when it's a list
+When(/^I have a system with "([^"]*)" as "([^"]*)" listed property$/) do |property_value, property_name|
+  steps %(
+    When I select the first system in the list
+    And I follow "Properties" in the content area
+    And I select "#{property_value}" from "#{property_name}"
+    And I click on "Update Properties"
+    Then I should see a "System properties changed" text
+    And I clean the search index on the server
+    And I wait for "5" seconds
+        )
+end
+
+# Confirm user has landed on a system's overview page
+Then(/^I should land on system's overview page$/) do
+  steps %(
+    Then I should see a "System Status" text
+    And I should see a "System Info" text
+    And I should see a "System Events" text
+    And I should see a "System Properties" text
+    And I should see a "Subscribed Channels" text
+        )
+end
+
+When(/^I enter minion hostname on the search field$/) do
+  step %(I enter "#{$minion.full_hostname}" on the search field)
+end
+
+Then(/^I should see minion hostname as first search result$/) do
+  hostname = $minion.full_hostname
```

Comment:
Well... why not, but we are just looking into attributes of a class, it should be fine to just call $minion.full_hostname a pair of times

Reply:
True, I just found it easier to write the variable name down in what we expect :thinking: